counsel.el

```diff
@@ -6312,6 +6327,19 @@ Use the presence of a \".git\" file to determine the root."
 Use the presence of a `dir-locals-file' to determine the root."
   (counsel--dominating-file dir-locals-file))
 
+(defun counsel--single-file-root ()
+  "Return the root of a single standalone file.
+
+To avoid catching un-related history for this one file we locally
+override `counsel-compile-history' and leave the user to fill in their
+preferred build stanza.
+
+This should be the last function in `counsel-compile-root-functions'
+after giving all the other project root finders a chance."
+  (unless (local-variable-p 'counsel-compile-history-exact-src)
+    (setq-local counsel-compile-history-exact-src 't))
```

Comment:
No need to quote `t`, and if you make `counsel-compile-history-exact-src` automatically buffer-local then you won't need the `setq-local`.